Add new pack - Hello Universe - 1.0.0

caroldelwing opened the pull request:
## Describe the Change

This PR includes the Hello Universe pack to the SpectroCloud community repo. This is a requirement of the upcoming PCG tutorial.
[Hello Universe](https://github.com/spectrocloud/hello-universe) is a demo web application utilized to help users learn more about Palette and its features.

karl-cardenas-coding commented on packs/hello-universe-1.0.0/manifests/hello-universe.yaml:
```diff
@@ -0,0 +1,33 @@
+    metadata:
+      labels:
+        app: hello-universe
+    spec:
```
We should specify CPU and memory resources in the `spec.containers` block. Try the below values because I'm pretty sure our container requires very few resources. You can see the resource utilization in a deployed cluster 😄
That way, we can also document it in the README. 

```
   resources:
      requests:
        cpu: "100m" # requesting 100 milliCPU
      limits:
        cpu: "200m" # limiting to 200 milliCPU
  ```

karl-cardenas-coding reviewed (changes requested):
Really great start and easy to follow. I just have some suggestions below that will make it easier for users to use the pack down the road 😄

caroldelwing commented on packs/hello-universe-1.0.0/logo.png:
I agree. I've utilized the Spectronaut image :astronaut:

caroldelwing commented on packs/hello-universe-1.0.0/manifests/hello-universe.yaml:
```diff
@@ -0,0 +1,33 @@
+metadata:
+  name: hello-universe-deployment
+spec:
+  replicas: 2
```
:white_check_mark:

caroldelwing commented on packs/hello-universe-1.0.0/manifests/hello-universe.yaml:
```diff
@@ -0,0 +1,33 @@
+        image: {{ .Values.registry }}/{{ .Values.repository }}:{{ .Values.tag }}
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 8080
```
The cluster port was exposed through the values YAML :white_check_mark:

caroldelwing commented on packs/hello-universe-1.0.0/pack.json:
```diff
@@ -0,0 +1,17 @@
+      "manifests/hello-universe.yaml"
+    ],
+    "layer": "addon",
+    "name": "hellouniverse",
```
:white_check_mark: Yes! :)

caroldelwing commented on packs/hello-universe-1.0.0/README.md:
```diff
@@ -0,0 +1,40 @@
+
+- A cluster profile where the Hello Universe pack can be integrated.
+
+- A Palette cluster with port `:8080` available.
```
:white_check_mark: Cluster port exposed and prereqs updated.

caroldelwing commented:
Here are a few updates: I've addressed the review comments, added the content section to the values.yaml file after talking to Vishwanath, included a schema.yaml file, and updated the manifest and values.yaml file so that they look similar to other packs. The Hello Universe pack was tested using a local registry and a VMware cluster, and all checks of this repo have passed. :)

karl-cardenas-coding commented on packs/hello-universe-1.1.1/README.md:
```diff
@@ -0,0 +1,42 @@
+# Hello Universe
+
+[Hello Universe](https://github.com/spectrocloud/hello-universe) is a demo web application utilized to help users learn more about [Palette](https://docs.spectrocloud.com/introduction) and its features. It functions as a standalone front-end application and provides users with a local click counter and funny SpectroCloud-themed images.
```
```suggestion
[Hello Universe](https://github.com/spectrocloud/hello-universe) is a demo web application utilized to help users learn more about [Palette](https://docs.spectrocloud.com/introduction) and its features. It functions as a standalone front-end application and provides users with a local click counter and funny Spectro Cloud-themed images.
```

karl-cardenas-coding commented on packs/hello-universe-1.1.1/manifests/hello-universe.yaml:
```diff
@@ -0,0 +1,45 @@
+            - containerPort: 8080
+          resources:
+            requests:
+              cpu: "100m" # requesting 100 milliCPU
```
In my example code snippet, I forgot to include memory. We should also define the memory 🙈

karl-cardenas-coding reviewed (dismissed):
Approved - Just one minor suggestion that I missed with my previous code snippet.
